src/logic/calculate.test.js

```diff
@@ -0,0 +1,8 @@
+import calculate from './calculate';
+
+describe('calculate', () => {
+  it('should return an object with total, next, and operation properties when passed an AC buttonName', () => {
+    const result = calculate({ total: '5', next: '3', operation: '+' }, 'AC');
+    expect(result).toEqual({ total: null, next: null, operation: null });
+  });
+});
```

Comment:
According to the requirements, tests that cover all scenarios of the calculate js should be written and I recommend writing more tests to see if other functions are working as expected.